resources/qml/PrintMonitor.qml

```diff
@@ -20,49 +20,462 @@ Column
+        }
+        Label
+        {
+            id: connectedPrinterAddressLabel
+            text: printerConnected ? connectedPrinter.address : ""
+            font: UM.Theme.getFont("small")
+            color: UM.Theme.getColor("text_inactive")
+            anchors.left: parent.left
+            anchors.leftMargin: UM.Theme.getSize("default_margin").width
+            anchors.top: parent.top
+            anchors.topMargin: UM.Theme.getSize("default_margin").height
+            anchors.right: parent.right
+            anchors.rightMargin: UM.Theme.getSize("default_margin").width
+            horizontalAlignment: Text.AlignRight
+        }
+        Label
+        {
             text: printerConnected ? connectedPrinter.connectionText : catalog.i18nc("@info:status", "The printer is not connected.")
             color: printerConnected && printerAcceptsCommands ? UM.Theme.getColor("setting_control_text") : UM.Theme.getColor("setting_control_disabled_text")
-            font: UM.Theme.getFont("default")
+            font: UM.Theme.getFont("very_small")
             wrapMode: Text.WordWrap
-            width: parent.width
+            anchors.left: parent.left
+            anchors.leftMargin: UM.Theme.getSize("default_margin").width
+            anchors.right: parent.right
+            anchors.rightMargin: UM.Theme.getSize("default_margin").width
+            anchors.top: connectedPrinterNameLabel.bottom
         }
     }
 
-    Loader
+    Rectangle
     {
-        sourceComponent: monitorSection
-        property string label: catalog.i18nc("@label", "Temperatures")
-    }
-    Repeater
-    {
-        model: machineExtruderCount.properties.value
-        delegate: Loader
+        color: UM.Theme.getColor("sidebar_lining")
+        width: parent.width
+        height: childrenRect.height
+
+        GridLayout
         {
-            sourceComponent: monitorItem
-            property string label: machineExtruderCount.properties.value > 1 ? extrudersModel.getItem(index).name : catalog.i18nc("@label", "Hotend")
-            property string value: printerConnected ? Math.round(connectedPrinter.hotendTemperatures[index]) + "°C" : ""
+            id: extrudersGrid
+            columns: 2
+            columnSpacing: UM.Theme.getSize("sidebar_lining_thin").width
+            rowSpacing: UM.Theme.getSize("sidebar_lining_thin").height
+            width: parent.width
+
+            Repeater
+            {
+                model: machineExtruderCount.properties.value
+                delegate: Rectangle
+                {
+                    id: extruderRectangle
+                    color: UM.Theme.getColor("sidebar")
+                    width: extrudersGrid.width / 2 - UM.Theme.getSize("sidebar_lining_thin").width / 2
+                    height: UM.Theme.getSize("sidebar_extruder_box").height
+                    Layout.fillWidth: index == machineExtruderCount.properties.value - 1 && index % 2 == 0
+
+                    Text //Extruder name.
```

Comment:
I was being consistent in what I thought was right: Label for labels, text for data.

Since you say Label works better if fonts can't be found, I'll change everything to use Label.